covid_graphs/covid_graphs/formula.py

```diff
@@ -2,42 +2,27 @@
 import math
 from abc import abstractmethod
 from dataclasses import dataclass
-from typing import Callable
+from typing import Callable, List, Tuple
 
 import numpy as np
 
 from . import fit_atg_model
 from .country_report import CountryReport
 
 
+@dataclass
 class Curve:
-    def __init__(
-        self,
-        func: Callable[[datetime.date], float],
-        start_date: datetime.date,
-        end_date: datetime.date,
-        label: str,
-    ) -> None:
-        """
-        Calculates function trace for dates in a half-open interval [start_date, end_date).
-
-        Sets the trace as properties `xs` and `ys`. Also calculates a point (x_max, y_max)
-        for which the maximum is achieved.
-
-        Note: The current design assumes the range of the plot is known at initialisation.
-        """
-        self.start_date = start_date
-        self.end_date = end_date
-        self.label = label
-
-        self.xs = [
-            start_date + datetime.timedelta(days=d) for d in range((end_date - start_date).days)
-        ]
-        self.ys = [func(x) for x in self.xs]
-
-        idx_max = np.argmax(self.ys)
-        self.x_max = self.xs[idx_max]
-        self.y_max = self.ys[idx_max]
+    func: Callable
+    start_date: datetime.date
+    display_at_least_until: datetime.date
```

Comment:
I find the dissymmetry between start an end suspicious. Is there a reason? What about using `display_at_least_from`? The function `generate_trace` would then take `start_date` as well.